Library/Homebrew/os/mac/cache_store.rb

```diff
@@ -0,0 +1,137 @@
+require "dbm"
```

Comment:
Noticing the filenames here: any reason to make this mac-only? `dbm` seems to be built by Ruby on Linux by default?

Reply:
The main reason I kept the file in the `os/mac/` directory is because the `LinkageChecker` is found in the `os/mac/` directory and the concrete `LinkageStore` in the `cache_store.rb` file is used by the linkage checker.

I can move the `cache_store.rb` file to a different location - since `dbm` is built into Ruby and Linux like you said. However, would you like me to break up the files and keep the `LinkageStore` class in the `os/mac/` directory since it's only being used by the `LinkageChecker`?

Reply:
> I can move the cache_store.rb file to a different location - since dbm is built into Ruby and Linux like you said. However, would you like me to break up the cache_store.rb file and keep the LinkageStore class in the os/mac/ directory since it's only being used by the LinkageChecker?

Sounds good 👍